rohmu/object_storage/config.py

```diff
@@ -83,10 +89,28 @@ class AzureObjectStorageConfig(StorageModel):
     account_key: Optional[str] = Field(None, repr=False)
     sas_token: Optional[str] = Field(None, repr=False)
     prefix: Optional[str] = None
+    is_secure: bool = True
+    host: Optional[str] = None
+    port: Optional[int] = None
     azure_cloud: Optional[str] = None
     proxy_info: Optional[ProxyInfo] = None
     storage_type: Literal[StorageDriver.azure] = StorageDriver.azure
 
+    @root_validator
+    @classmethod
+    def host_and_port_must_be_set_together(cls, values: Dict[str, Any]) -> Dict[str, Any]:
+        if "host" in values and "port" in values:
```

Comment:
This test is redundant because the fields have a default value and so they are always present. BTW, another option would be to introduce a `HostInfo` model, similar to `ProxyInfo`, containing the new fields. This way there would be no need for a custom validator, and it would be more consistent. WDYT?

Reply:
Good catch, fixing that on the latest commit. 

About HostInfo, I like the idea, but we could do that as a separate refactoring PR as there are many places where this type would fit. At the moment this is consistent with the S3 approach to set custom host/port.

Reply:
Agreed, I did not realize that this is based on the existing S3 config.